merge/default_keys_test.go

```diff
@@ -322,6 +322,61 @@ func TestDefaultKeysNested(t *testing.T) {
 				),
 			},
 		},
+		"apply_default_type_differs": {
```

Comment:
Perhaps a comment or make the name clear that it's float/int that you're testing?